libraries/abstractions/common_io/test/test_iot_i2c.c

```diff
@@ -378,136 +390,221 @@ TEST( TEST_IOT_I2C, AFQP_IotI2CReadAsyncSuccess )
+        lRetVal = iot_i2c_ioctl(xI2CHandle, eI2CSetSlaveAddr, &uctestIotI2CSlaveAddr );
+        TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
 
-    /* Set i2c not stop between transaction */
-    lRetVal = iot_i2c_ioctl(xI2CHandle, eI2CSendNoStopFlag, NULL);
-    TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
+        /* Set i2c not stop between transaction */
+        lRetVal = iot_i2c_ioctl(xI2CHandle, eI2CSendNoStopFlag, NULL);
+        TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
 
-    /* write the device register address. */
-    lRetVal = iot_i2c_write_sync(xI2CHandle, &xtestIotI2CDeviceRegister, sizeof(xtestIotI2CDeviceRegister));
-    TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
+        /* write the device register address. */
+        lRetVal = iot_i2c_write_sync(xI2CHandle, &xtestIotI2CDeviceRegister, sizeof(xtestIotI2CDeviceRegister));
+        TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
 
-    /* Set i2c slave address */
-    lRetVal = iot_i2c_ioctl(xI2CHandle, eI2CSetSlaveAddr, &uctestIotI2CSlaveAddr );
-    TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
+        /* Set i2c slave address to read from the device register. */
+        lRetVal = iot_i2c_ioctl(xI2CHandle, eI2CSetSlaveAddr, &uctestIotI2CSlaveAddr );
+        TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
 
-    /* Set i2c not stop between transaction */
-    lRetVal = iot_i2c_ioctl(xI2CHandle, eI2CSendNoStopFlag, NULL);
-    TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
+        /* Read from i2c device for single byte. */
+        lRetVal = iot_i2c_read_async(xI2CHandle, &singleByte, sizeof(singleByte));
+        TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
 
-    /* read from i2c device register */
-    lRetVal = iot_i2c_read_async(xI2CHandle, ret_val, sizeof(ret_val));
-    TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
+        lRetVal = xSemaphoreTake(xtestIotI2CSemaphore, testIotI2C_MAX_TIMEOUT);
+        TEST_ASSERT_EQUAL(pdTRUE, lRetVal);
 
-    /* Set i2c congifuration during transaction */
-    lRetVal = iot_i2c_ioctl(xI2CHandle, eI2CSetMasterConfig, &xI2CConfig);
-    TEST_ASSERT_EQUAL(IOT_I2C_BUSY, lRetVal);
+        lRetVal = iot_i2c_ioctl(xI2CHandle, eI2CGetRxNoOfbytes, &readBytes);
+        TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
+        /* Assert the number of bytes being read is 1. */
+        TEST_ASSERT_EQUAL(sizeof(singleByte), readBytes);
 
-    lRetVal = xSemaphoreTake(xtestIotI2CSemaphore, testIotI2C_MAX_TIMEOUT);
-    TEST_ASSERT_EQUAL(pdTRUE, lRetVal);
+        lRetVal = iot_i2c_ioctl(xI2CHandle, eI2CGetTxNoOfbytes, &writeBytes);
+        TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
+        /* Assert the number of bytes being written is 1. */
+        TEST_ASSERT_EQUAL(sizeof(xtestIotI2CDeviceRegister), writeBytes);
+    }
 
     lRetVal = iot_i2c_close(xI2CHandle);
     TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
 }
 
 /*-----------------------------------------------------------*/
 
+typedef struct CallbackParam {
```

Comment:
I think this should go at the top of the file

Reply:
ok